network/network-tutorial-overlay.md

```diff
@@ -298,154 +298,129 @@ ports must be open between the two Docker hosts:
-    6af747d9ae1e        none                null                local
-    uw9etrdymism        test-net            overlay             swarm
+    ...
+    ytqk46ddg30p        test-net            overlay             swarm
     ```
 
-5.  Remember that you created `alpine1` from `host1` and `alpine2` from `host2`.
-    Now, attach to `alpine2` from `host1`:
+7.  On `host1`, ping `alpine2` within the interactive terminal of `alpine1`:
 
     ```bash
-    (host1) $ docker container attach alpine2
+    / # ping -c 2 alpine2
+    PING alpine2 (10.0.0.5): 56 data bytes
+    64 bytes from 10.0.0.5: seq=0 ttl=64 time=0.600 ms
+    64 bytes from 10.0.0.5: seq=1 ttl=64 time=0.555 ms
 
-    #
+    --- alpine2 ping statistics ---
+    2 packets transmitted, 2 packets received, 0% packet loss
+    round-trip min/avg/max = 0.555/0.577/0.600 ms
     ```
 
-    Automatic service discovery worked between two containers across the overlay
-    network!
+    The two containers can communicate using the overlay network connecting
+    `host1` and `host2`. If you run another alpine container on `host2` that is
+    not detached (say, `alpine3`), you can ping `alpine1` from `host2`.
 
-    Within the attached session, try pinging `alpine1` from `alpine2`:
+8.  On `host1`, close the `alpine1` session (which also stops the container):
 
     ```bash
-    # ping -c 2 alpine1
-
-    PING alpine1 (10.0.0.2): 56 data bytes
-    64 bytes from 10.0.0.2: seq=0 ttl=64 time=0.523 ms
-    64 bytes from 10.0.0.2: seq=1 ttl=64 time=0.547 ms
-
-    --- alpine1 ping statistics ---
-    2 packets transmitted, 2 packets received, 0% packet loss
-    round-trip min/avg/max = 0.523/0.535/0.547 ms
+    / # exit
     ```
 
-    This proves that the two containers can communicate with each other using
-    the overlay network which is connecting `host1` and `host2`.
+9.  Clean up your containers and networks:
 
-    Detach from `alpine2` using the `CTRL` + `P` `CTRL` + `Q` sequence.
+    Because the Docker daemons operate independently and these are standalone
+    containers, you must run the commands on the individual hosts.
 
-6.  Stop the containers and remove `test-net` from each host. Because the Docker
-    daemons are operating independently and these are standalone containers, you
-    need to run the commands on the individual hosts.
+    a.  On `host1`, remove `alpine1` and `test-net`:
 
     ```bash
-    (host1) $ docker container stop alpine1
-            $ docker container rm alpine1
-            $ docker network rm test-net
+    $ docker container rm alpine1
+    $ docker network rm test-net
     ```
 
+    b.  On `host2`, stop and remove `alpine2` and  remove `test-net`:
+
     ```bash
-    (host2) $ docker container stop alpine2
-            $ docker container rm alpine2
-            $ docker network rm test-net
+    $ docker container stop alpine2
+    $ docker container rm alpine2
+    $ docker network rm test-net
```

Comment:
Removal of test-net not necessary.
Command results in Error: No such network: test-net

Reply:
@fheuer , please test the new instructions:

1.  Set up a Docker swarm with 1 manager (`host1`) and 1 worker (`host2`).
2.  On `host1`, create an attachable overlay network (`test-net`) and run an interactive alpine container (`alpine1): 

    ```
    $ docker network create --driver=overlay --attachable test-net
    $ docker run -it --name alpine1 --network test-net alpine
    / #
    ```

3.  On `host2`, start a detached and interactive alpine container (`alpine2`) that connects to `test-net`, then list the networks to ensure the overlay network (`test-net`) was created:

    ```
    $ docker run -dit --name alpine2 --network test-net alpine
    $ docker network ls

    ```
4.  On `host1`, ping `alpine2` within the interactive terminal of `alpine1`:

    ```
    / # ping -c 2 alpine2
    ```

Reply:
Yep works! Still the issue stated above though.

Reply:
Ah ha. I see what you're saying. Thanks for the feedback @fheuer